pkg/resource/function/hooks.go

```diff
@@ -114,24 +113,17 @@ func (rm *resourceManager) customUpdateFunction(
 	// UpdateFunctionCode because both of them can put the function in a
 	// Pending state.
 	switch {
-	case delta.DifferentAt("Spec.Code"):
+	case delta.DifferentAt("Spec.Code") || delta.DifferentAt("Spec.Code.SHA256"):
 		err = rm.updateFunctionCode(ctx, desired, delta)
```

Comment:
Another question that I have is: what if the user provides the wrong sha256? how does the controller behave?